src/Network/Http/Client.php

```diff
@@ -380,7 +380,8 @@ public function buildUrl($url, $query = [], $options = [])
         }
         if ($query) {
             $q = (strpos($url, '?') === false) ? '?' : '&';
-            $url .= $q . http_build_query($query);
+            $url .= $q;
+            $url .= is_string($query) ? $query : http_build_query($query);
```

Comment:
unrelated to the PR, but http_build_query doesn't exist if you use pecl-http v2 and up, its been completely redone to be OOP style